rocky/reports/report_types/definitions.py

```diff
@@ -107,6 +107,34 @@ def to_hostnames(self, input_oois: Iterable[str], valid_time: datetime) -> dict[
 
         return hostnames_by_input_ooi
 
+    def to_human_readable_hostnames(
+        self,
+        input_oois: Iterable[str],
+        valid_time: datetime,
+        extra_indicator: str = ", ...",
+        open_indicator: str = "(",
+        close_indicator: str = ")",
+        hostnames_by_input_ooi: dict | None = None,
+    ) -> dict[str, str]:
+        """Converts input_oois to human readable hostname strings.
+
+        Turns a list of either Hostname and IPAddress references into a string
+        of shortest related hostname and indication on more hostnames present,
+        grouped by input ooi.
+        """
+        if hostnames_by_input_ooi is None:
+            hostnames_by_input_ooi = self.to_hostnames(input_oois, valid_time)
+        human_readable = {}
+        for input_ooi, hostnames in hostnames_by_input_ooi.items():
+            hostname_str = ""
+            if hostnames:
+                hostnames = [h.human_readable for h in hostnames]
+                add_str = extra_indicator if len(hostnames) > 1 else ""
+                hostname_str = open_indicator + min(hostnames, key=len) + add_str + close_indicator
+            human_readable[input_ooi] = hostname_str
```

Comment:
I think writing it like this makes it easier to understand. I didn't test the code.

```suggestion
            if hostnames:
                hostname = min([h.human_readable for h in hostnames], key=len)
                if len(hostnames) > 1:
                    human_readable[input_ooi] = f"({hostname}, ...)"
                else:
                    human_readable[input_ooi] = f"({hostname})"
            else:
                human_readable[input_ooi] = ""
```

Reply:
> I actually prefer a single assignment to the `human_readable` dictionary, as that keeps the code more maintainable in my experience. However, line 131 is indeed redundant as the length does not need to be determined from the human readable strings.
> 
> Or maybe the `add_str` and `hostname_str` could be combined in a single f-string expression

Your suggestion looks fine. I'm not sure if an f-string would make it more readable in this case; but I'll leave that up to you.

Reply:
> I actually prefer a single assignment to the `human_readable` dictionary, as that keeps the code more maintainable in my experience. However, line 131 is indeed redundant as the length does not need to be determined from the human readable strings.

I find the way you wrote the method is a lot harder to understand than is necessary and thus harder to maintain. They way I changed it is very clear that there are three different cases and what the string in those three different cases would be. With your code there are 6 more variables you need to keep track of in your head to understand what this does.

Reply:
I have now rewritten the whole method